docs/proposals/20210222-kubelet-authentication.md

```diff
@@ -0,0 +1,376 @@
+- Ensure that this can work with any infrastructure provider
+
+
+### Non-Goals/Future Work
+
+- To change assumptions around management cluster to workload cluster connectivity
+- Solve the protection of initial cluster bootstrap secrets for the control plane nodes
+- To be a mandatory requirement of using Cluster API
+- To implement or enable hardware-backed encryption of traffic between worker nodes and the control plane components.
+
+## Proposal
+
+### User Stories
+
+#### Story 1: machine attestation
+
+A cluster operator has been asked to ensure compliance with [NIST SP 800-190 Application Container Security][nist-sp-800-190] Guide. Hardware countermeasure 4.6 suggests that container platforms should make use of trusted computing. In a Kubernetes context, this would mean providing hardware node attestation wherever possible.
+
+### Requirements
+
+The node bootstrapper MUST be able to attest the identity of the machine against a chain of trust provided by the hardware or cloud provider.
+
+
+### Implementation Details/Notes/Constraints
+
+
+| Component                   | Location                                                                                                                                                                                                                                                                                                                                         | Description                                                                                                                                                                                                                                                                |
+| --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| node-attestation-controller | Part of Cluster API repository, under bootstrap/node/attestation, and imported by Cluster API infra providers for implementation. A generic challenge-response implementation will be included for providers / bare metal without an attestation mechanism. This controller runs as part of the cluster API components in the management cluster | A controller to verify and sign the CSR. This would be typically an importable controller where the infrastructure provider implements the interface with specific code for CSR approval and start the controller as part of its main.go or through an independent binary. |
```

Comment:
```suggestion
| node-attestation-controller | Part of Cluster API repository, under bootstrap/node/attestation, and imported by Cluster API infra providers for implementation. A generic challenge-response implementation will be included for providers / bare metal without an attestation mechanism. This controller runs as part of the Cluster API components in the management cluster | A controller to verify and sign the CSR. This would be typically an importable controller where the infrastructure provider implements the interface with specific code for CSR approval and start the controller as part of its main.go or through an independent binary. |
```

Reply:
Incredible work parsing the markdown table